### PR TITLE
282 update wiki

### LINKED ISSUE #282: Update Wiki
As a developer, I'd like to have an updated wiki, in order to only find relevant information about the project quickly.

---

### :thought_balloon: &nbsp; Hints & Thoughts
Check popular open-source projects and steal some quick win ideas we can implement

### :question: &nbsp; Additional Questions to Answer / Consider

---

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] Should delete files or parts of files that are not needed anymore (have a look at hints and thoughts) -> make sure you extract all the information from these files that is still useful (rule of thumb: everything you can just google is not relevant)
- [ ] Add diagrams from the technical presentation to the wiki
- [ ] Should enrich the architecture text with pointers to the code files or folders that implement specific elements of the architecture.
- [ ] Refactor the structure and naming of the pages in the wiki if needed 
- [ ] Check all final wiki files for typos.

### PR DESCRIPTION
**Pull request by lukashueller**
In this PR we revised the wiki. That includes the following steps:
- delete all files, whose content can also be obtained from the Internet through research
- update the documentation of our concepts
- updating the sidebar and start page of the wiki
- testing the new wiki incl. all links!

closes #282

### CODE REVIEW

File: wiki/Concept-Starting-and-Orchestrating-Bots.md [hunk 1]
```diff
@@ -1,89 +0,0 @@
...
-
-### Higher Abstraction Frameworks
-
-There are some frameworks using one (or more) of these basic approaches but are one abstraction layer higher for the user. In particular, pusher and socket.io were looked at.
-Pusher is based on a publish subscribe approach simular to SSE and it seems to be very easy to use. The biggest disadvantage was that it is not open-source but ultimatly one would have to pay for it. Socket.io is a highly popular (> 4 Mio. weekly npm downloads) open source framework. It is mainly based on websocket communication but is also able to seamlessly switch to a long polling approach in the background if websockets do not work.
-
-### Decision
-
-In the end, we decided for socket.io as it is open-cource, well-supported, 'easy' to use, robust and websocket based enabling a bidirectional communication.
-To get started it is recommended reading [this](https://socket.io/docs/v4/index.html) introduction to socket.io. Especially these two subpages [1](https://socket.io/docs/v4/server-socket-instance/) & [2](https://socket.io/docs/v4/client-socket-instance/) are relevant for this usecase.
```
Comment (WolfgangDaniel) [inline wiki/Concept-Starting-and-Orchestrating-Bots.md:75]: I would leave that part on how we decided on the communication with the desktop app :)

File: wiki/Coding-Standards.md [hunk 2]
```diff
@@ -67,12 +67,15 @@ Another Code Formatting Tool being used is [Prettier](https://prettier.io/). Her
 - "Single Source of Truth" is abbreviated with `Ssot` or `ssot` (don't use `SSoT` or `SSOT`)
 - use `Robot` in all cases instead of `Bot`
 - use `Application` in all cases instead of `App` (in context of supported RPA Applications)
+- use **hyphens** for CSS-classes and CSS-ids consistently
+  For example, don't call the class `buttonBackground` and instead call it `button-background`.
```
Comment (SanJSp) [inline wiki/Coding-Standards.md:71]: There is either a dot missing or more indentation is needed.

```suggestion
- use **hyphens** for CSS-classes and CSS-ids consistently
   - For example, don't call the class `buttonBackground` and instead call it `button-background`.
```
Before:
grafik

After: 
grafik

File: wiki/How-To-Style-Using-CSS.md [hunk 3]
```diff
@@ -1,415 +0,0 @@
...
-
-- naming:
-  For the naming of classes and ids please use **hyphens** consistently.
-  For example, don't call the class `buttonBackground` and instead call it `button-background`.
-- sizing:
-  Try to use only relative units (vw,vh,rem,%) to size elements and **not** absolut units (px)
-
-</p></details>
-
----
```
Comment (WolfgangDaniel) [inline wiki/How-To-Style-Using-CSS.md:402]: I would leave this part as it is focussed on how we want to use styling in the project and is not just a basic explaining section.

**Review by WolfgangDaniel — CHANGES_REQUESTED** [review]

File: wiki/Concept-Starting-and-Orchestrating-Bots.md [hunk 4]
```diff
@@ -1,89 +0,0 @@
...
-
-**Database/Server:**
-We implemented a jobs collection in MongoDB as well as a Mongoose jobs model. Every job has a robotId, a userId, a status (waiting/executing/success/failed) and an array of parameters that contains the arguments the user entered in the web frontend when starting the robot execution.
-
-**Server:**
-Sets up a server and socket instance, establishes socket connection with the web frontend and the desktop app, groups sockets by userIds (by using the room concept), reacts on robot execution commands and forwards this command to the desktop app and updates the jobs collection in MongoDB continiously.
-
-**Web Frontend:**
-Connects with the server using a socket connection. Also, like the dektop app, we join a userId specific room whenever the robot overview is rendered. Additionally, we send a robot execution job to the backend when the user clicks on the play button in the robot container.
```
Comment (SanJSp) [inline wiki/Concept-Starting-and-Orchestrating-Bots.md:89]: I think a part regarding how the socket.io implementation works would have been quite nice. The local client readme does not explain it with as much detail like here and it would for sure help users understand the implementation better. But seeing the time this is out of scope.

File: wiki/Documentation-Single-Source-of-Truth.md [hunk 5]
```diff
@@ -77,7 +77,7 @@ No intermediate events are currently supported.
 
 ### Overview SSOT hierarchy
 
-![](https://i.imgur.com/7yOqDZv.png)
+![image](https://user-images.githubusercontent.com/36270527/120194098-705e4b00-c21d-11eb-9301-937447369e93.png)
```
Comment (SanJSp) [inline wiki/Documentation-Single-Source-of-Truth.md:80]: Can you have a look at the heading of `**parameter object extending the INSTRUCTION:**`? It's a few lines below. I'd like to change it to:

`**Parameter Object Extending the INSTRUCTION Element**`

File: wiki/Home.md [hunk 6]
```diff
@@ -1,29 +1,23 @@
...
-- documentation the [corporate identity of Ark_automate](https://github.com/bptlab/ark_automate/wiki/Documentation-Corporate-Identity)
-- documentation regarding our [team work](https://github.com/bptlab/ark_automate/wiki/Team-Scrum-process)
-- concepts on how we think users to interact with [the bots](https://github.com/bptlab/ark_automate/wiki/concept-Starting-and-orchestrating-Bots)
-- our own guides to learn how to
-  - write [code documentation](https://github.com/bptlab/ark_automate/wiki/How-to-write-code-documentation)
-  - write [tests](https://github.com/bptlab/ark_automate/wiki/How-to-write-tests)
-  - sytle [our components using css](https://github.com/bptlab/ark_automate/wiki/How-to-style-using-css)
+- our [vision](https://github.com/bptlab/ark_automate/wiki/Vision-for-Ark-automate) for this project including a pitch, the architecture and the user interaction
+- documentation regarding [the structure](https://github.com/bptlab/ark_automate/wiki/Documentation-Folder-Structure) of this project
+- our [code documentation](https://bptlab.github.io/ark_automate/) hostet on GitHub Pages grouped by modules and classes
```
Comment (SanJSp) [inline wiki/Home.md:9]: I'd propose a different order here

```suggestion
- our [code documentation](https://bptlab.github.io/ark_automate/) hostet on GitHub Pages grouped by modules and classes
- documentation regarding [the structure](https://github.com/bptlab/ark_automate/wiki/Documentation-Folder-Structure) of this project
```

File: wiki/_Sidebar.md [hunk 7]
```diff
@@ -1,24 +1,16 @@
...
-
-- Concepts
-    - [Planned user interaction](https://github.com/bptlab/ark_automate/wiki/concept-Starting-and-orchestrating-Bots) in the summer of 2021
-
-- Guides
-    - [Code documentation](https://github.com/bptlab/ark_automate/wiki/How-to-write-code-documentation)
-    - [Tests](https://github.com/bptlab/ark_automate/wiki/How-to-write-tests)
-    - [Style our components using css](https://github.com/bptlab/ark_automate/wiki/How-to-style-using-css)
+    - [Database Communication](https://github.com/bptlab/ark_automate/wiki/Github-Workflows/Database-and-Communication)
+    - [Testing Conventions](https://github.com/bptlab/ark_automate/wiki/Github-Workflows/Testing-Conventions)
```
Comment (SanJSp) [inline wiki/_Sidebar.md:16]: These links lead to the wrong page. I guess you meant the proposed sites.

```suggestion
    - [Database Communication](https://github.com/bptlab/ark_automate/wiki/Database-and-Communication)
    - [Testing Conventions](https://github.com/bptlab/ark_automate/wiki/Testing-Conventions)
```

**Review by SanJSp — APPROVED** [review]
Good job 😊 just some small fixes. As I think you can fix them yourself, I approve with these changes ☺️

File: wiki/How-To-Style-Using-CSS.md (hunk 3 above)
Comment (lukashueller) [reply wiki/How-To-Style-Using-CSS.md:402]: fair point!

File: wiki/Concept-Starting-and-Orchestrating-Bots.md (hunk 1 above)
Comment (lukashueller) [reply wiki/Concept-Starting-and-Orchestrating-Bots.md:75]: I think the decision for Socket.io and also the documentation of the implementation is important. I have added these two points directly again. However, I find the decision compared to the other communication methods not worth documenting or uninteresting for newcomers. What do you think about it @SanJSp ?

File: wiki/Concept-Starting-and-Orchestrating-Bots.md (hunk 4 above)
Comment (lukashueller) [reply wiki/Concept-Starting-and-Orchestrating-Bots.md:89]: reimplemented!

File: wiki/Concept-Starting-and-Orchestrating-Bots.md (hunk 1 above)
Comment (SanJSp) [reply wiki/Concept-Starting-and-Orchestrating-Bots.md:75]: I think the pages are irrelvant, if we want to keep something, then it would be this:

> In the end, we decided for socket.io as it is open-cource, well-supported, 'easy' to use, robust and websocket based enabling a bidirectional communication.

We could add it to the re-added socket.io description. See [this ](https://github.com/bptlab/ark_automate/pull/333#discussion_r642558813) comment

Comment (lukashueller) [reply wiki/Concept-Starting-and-Orchestrating-Bots.md:75]: Perfect Sandro! Exactly these lines I have already copied :D

File: wiki/Documentation-Communication-Local-Client.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# How the approach with socket.io was implemented our platform
```
Comment (SanJSp) [inline wiki/Documentation-Communication-Local-Client.md:1]:
```suggestion
# How the approach with socket.io was implemented on our platform
```

File: wiki/Documentation-Communication-Local-Client.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+**Server:**
+Sets up a server and socket instance, establishes socket connection with the web frontend and the desktop app, groups sockets by userIds (by using the room concept), reacts on robot execution commands and forwards this command to the desktop app and updates the jobs collection in MongoDB continiously.
+
+**Web Frontend:**
+Connects with the server using a socket connection. Also, like the dektop app, we join a userId specific room whenever the robot overview is rendered. Additionally, we send a robot execution job to the backend when the user clicks on the play button in the robot container.
+
+## Why we Use Socket.io
+
+In the end, we decided for socket.io as it is open-cource, well-supported, 'easy' to use, robust and websocket based enabling a bidirectional communication.
+To get started it is recommended reading [this](https://socket.io/docs/v4/index.html) introduction to socket.io. Especially these two subpages [1](https://socket.io/docs/v4/server-socket-instance/) & [2](https://socket.io/docs/v4/client-socket-instance/) are relevant for this usecase.
```
Comment (SanJSp) [inline wiki/Documentation-Communication-Local-Client.md:18]: This line can in my opinion be removed

**Review by SanJSp — APPROVED** [review]
Thanks for implementing my proposed changes. I've got minor, unimportant things to add.

**Review by WolfgangDaniel — APPROVED** [review]
Nice Job, thank you! :D
